### PR TITLE
RavenDB-13367 LinQ to CreateIndex generating invalid C# code in runtime

### PR DESCRIPTION
**Pull request by ayende**

### CODE REVIEW

**Comment by ppekrol** [comment]
@ayende can you review my change? Also I think this should go to 4.2 only?

File: src/Raven.Client/Documents/Indexes/ExpressionStringBuilder.cs [hunk 1]
```diff
@@ -171,13 +172,49 @@ private void OutMember(Expression instance, MemberInfo member, Type exprType)
...
+            if (name == null)
+                return false;
+
+            if (name.Length > 512 || name.Length <= 0)
+                return false;
+
+            if (Regex.IsMatch(name, @"<>([a-z])__TransparentIdentifier(\w+)"))
+                return true; // we are transforming those to 'thisX' where X is extracted from the name
+
+            if (char.IsLetter(name[0]) == false)
```
Comment (ayende) [inline src/Raven.Client/Documents/Indexes/ExpressionStringBuilder.cs:207]:
```suggestion
            if (char.IsLetter(name[0]) == false && name[0] != '_')
```

**Review by ayende — COMMENTED** [review]
I'm fine with making this 4.2 only

**Comment by ppekrol** [comment]
4.2: https://github.com/ravendb/ravendb/pull/9178

**Pull request closed without merging**